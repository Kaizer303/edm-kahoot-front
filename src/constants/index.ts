export const LOCALE = {
  TH: 'th',
  EN: 'en',
}

export const SUPPORTED_LOCALES = [LOCALE.TH, LOCALE.EN]