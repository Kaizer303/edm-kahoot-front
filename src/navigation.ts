import { createSharedPathnamesNavigation } from 'next-intl/navigation'
import { SUPPORTED_LOCALES } from './constants'

export const locales = SUPPORTED_LOCALES
export const localePrefix = 'always'

export const { Link, redirect, usePathname, useRouter } = createSharedPathnamesNavigation({
  locales,
  localePrefix,
})