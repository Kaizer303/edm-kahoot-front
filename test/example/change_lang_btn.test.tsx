import ChangeLangBtn from '@/components/example/change_lang_btn'
import { fireEvent, render, screen } from '@testing-library/react'
import { useRouter } from '@/navigation'
import { LOCALE } from '@/constants'
import { useLocale } from 'next-intl'

jest.mock('@/navigation', () => ({
  usePathname: jest.fn().mockReturnValue('/mocked-pathname'),
  useRouter: jest.fn().mockReturnValue({
    replace: jest.fn(),
  }),
}))

jest.mock('next-intl', () => ({
  useLocale: jest.fn().mockReturnValue('th'),
  useTranslations: jest.fn().mockReturnValue(
    jest.fn().mockReturnValue('button_label'),
  ),
}))

describe('[Unit Test] components/example/change_lang_btn.tsx', () => {
  describe('[Snapshot Test]', () => {
    it('should match snapshot', () => {
      const { asFragment } = render(<ChangeLangBtn />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('[Component Test]', () => {
    it('should called hooks with correctly pathname and locale EN', async () => {
      const mockRouter = {
        replace: jest.fn(),
      }

      const mockedRouter = useRouter as jest.Mock
      mockedRouter.mockReturnValue(mockRouter)

      const mockedLocale = useLocale as jest.Mock
      mockedLocale.mockReturnValue(LOCALE.TH)

      render(<ChangeLangBtn />)

      fireEvent.click(screen.getByRole('button'))
      expect(mockRouter.replace).toHaveBeenCalledWith('/mocked-pathname', { locale: LOCALE.EN })
    })

    it('should called hooks with correctly pathname and locale TH', async () => {
      const mockRouter = {
        replace: jest.fn(),
      }

      const mockedRouter = useRouter as jest.Mock
      mockedRouter.mockReturnValue(mockRouter)

      const mockedLocale = useLocale as jest.Mock
      mockedLocale.mockReturnValue(LOCALE.EN)

      render(<ChangeLangBtn />)

      fireEvent.click(screen.getByRole('button'))
      expect(mockRouter.replace).toHaveBeenCalledWith('/mocked-pathname', { locale: LOCALE.TH })
    })
  })
})