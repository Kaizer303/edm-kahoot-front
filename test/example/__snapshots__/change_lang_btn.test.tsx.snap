// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Unit Test] components/example/change_lang_btn.tsx [Snapshot Test] should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-mzwlov ant-btn-primary"
    type="button"
  >
    <span>
      button_label
    </span>
  </button>
</DocumentFragment>
`;
